ospf6d/ospf6_asbr.c

```diff
@@ -2613,3 +2693,980 @@ void install_element_ospf6_debug_asbr(void)
+		}
+
+		UNSET_FLAG(aggr->aggrflags, OSPF6_EXTERNAL_AGGRT_ORIGINATED);
+
+		if (IS_OSPF6_DEBUG_AGGR)
+			zlog_debug("%s: Don't originate the summary address,It is configured to not-advertise.",
+				__func__);
+		return;
+	}
+
+	/* Summary route already originated,
+	 * So, Do nothing.
+	 */
+	if (CHECK_FLAG(aggr->aggrflags, OSPF6_EXTERNAL_AGGRT_ORIGINATED)) {
+		if (!aggr_lsa) {
+			zlog_warn(
+				"%s: Could not refresh/originate %pFX",
+						__func__,
+						&aggr->p);
+			/* Remove the assert later */
+			assert(aggr_lsa);
+			return;
+		}
+
+		external = (struct ospf6_as_external_lsa *)OSPF6_LSA_HEADER_END
+					(aggr_lsa->header);
+		metric = (unsigned long)OSPF6_ASBR_METRIC(external);
+		tag = ospf6_as_external_lsa_get_tag(aggr_lsa);
+		mtype = CHECK_FLAG(external->bits_metric,
+				   OSPF6_ASBR_BIT_E) ? 2 : 1;
+
+		/* Prepare the external_info for aggregator */
+		ospf6_fill_aggr_route_details(ospf6, aggr);
+		rt_aggr = aggr->route;
+		/* If tag/metric/metric-type modified , then re-originate the
+		 * route with modified tag/metric/metric-type details.
+		 */
+		if ((tag != aggr->tag)
+		    || (metric != (unsigned int)rt_aggr->path.cost)
+		    || (mtype != aggr->mtype)) {
+
+			if (IS_OSPF6_DEBUG_AGGR)
+				zlog_debug(
+					"%s: Routetag(old:%d new:%d)/Metric(o:%u,n:%u)/mtype(o:%d n:%d) modified,So refresh the summary route.(%pFX)",
+					__func__, tag, aggr->tag,
+					metric,
+					aggr->metric,
+					mtype, aggr->mtype,
+					&aggr->p);
+
+			aggr_lsa = ospf6_originate_type5_type7_lsas(aggr->route,
+								    ospf6);
+			if (aggr_lsa)
+				SET_FLAG(aggr->aggrflags,
+					OSPF6_EXTERNAL_AGGRT_ORIGINATED);
+		}
+
+		return;
+	}
+
+	/* If the external route prefix same as aggregate route
+	 * and if external route is already originated as TYPE-5
+	 * then it need to be refreshed and originate bit should
+	 * be set.
+	 */
+	if (lsa && prefix_same((struct prefix *)&aggr->p,
+				(struct prefix *)&rt->prefix)) {
+		if (IS_OSPF6_DEBUG_AGGR)
+			zlog_debug("%s: External route prefix is same as aggr so refreshing LSA(%pFX)",
+				__PRETTY_FUNCTION__,
+				&aggr->p);
+
+		THREAD_OFF(lsa->refresh);
+		thread_add_event(master, ospf6_lsa_refresh, lsa, 0,
+				 &lsa->refresh);
+		aggr->id = info->id;
+		SET_FLAG(aggr->aggrflags, OSPF6_EXTERNAL_AGGRT_ORIGINATED);
+		return;
+	}
+
+	ospf6_originate_new_aggr_lsa(ospf6, aggr);
+}
+
+static void ospf6_aggr_handle_external_info(void *data)
+{
+	struct ospf6_route *rt = (struct ospf6_route *)data;
+	struct ospf6_external_aggr_rt *aggr = NULL;
+	struct ospf6_lsa *lsa = NULL;
+	struct ospf6_external_info *info;
+	struct ospf6 *ospf6 = NULL;
+	struct prefix prefix_id;
+	struct route_node *node;
+
+	rt->aggr_route = NULL;
+
+	rt->to_be_processed = true;
+
+	if (IS_OSPF6_DEBUG_ASBR || IS_OSPF6_DEBUG_ORIGINATE(AS_EXTERNAL))
+		zlog_debug("%s: Handle external route for origination/refresh (%pFX)",
+					__func__,
+					&rt->prefix);
+
+	ospf6 = ospf6_lookup_by_vrf_name(VRF_DEFAULT_NAME);
```

Comment:
Please make this VRF-aware. This will not work on VRFs that are not the default.

Reply:
I have raised an issue #9058 to make this feature VRF aware by adding vrf keyword in CLI and taking care of other changes. This will be taken care as part of a separate PR.

Reply:
I have made the changes for supporting this feature in vrfs as well.